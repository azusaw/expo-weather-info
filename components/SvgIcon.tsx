import React from "react";
import Svg from "react-native-svg";
import { IconPaths } from "@/constants/IconPaths";

export enum IconType {
  Drop = "drop",
  Rain = "rain",
  Wind = "wind",
}

type IconProps = {
  name: IconType;
  size?: number;
  color?: string;
};

const getIconPath = (name: IconType) => {
  switch (name) {
    case IconType.Drop:
      return IconPaths.drop;
    case IconType.Rain:
      return IconPaths.rain;
    case IconType.Wind:
      return IconPaths.wind;
    default:
      return null;
  }
};

const SvgIcon = ({ name, size = 100, color = "#888" }: IconProps) => {
  const path = getIconPath(name);
  return (
    path && (
      <Svg viewBox="0 0 570 570" width={size} height={size} fill={color}>
        {path}
      </Svg>
    )
  );
};

export default SvgIcon;
