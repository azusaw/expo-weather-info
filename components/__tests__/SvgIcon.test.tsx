import { render } from "@testing-library/react-native";
import SvgIcon, { IconType } from "@/components/SvgIcon";

const mockProps = {
  name: IconType.Drop,
};

describe("SvgIcon", () => {
  test("it renders SvgIcon successfully", () => {
    const { toJSON } = render(<SvgIcon {...mockProps} />);
    expect(toJSON()).toMatchSnapshot();
  });

  Object.values(IconType).forEach((icom) =>
    test(`it renders ${icom} icon successfully`, () => {
      const component = render(<SvgIcon name={icom} />);
      expect(component).toBeTruthy();
    }),
  );

  test("it renders null when name is not match", () => {
    // @ts-ignore
    const { toJSON } = render(<SvgIcon name="hoge" />);
    expect(toJSON()).toBeNull();
  });
});
