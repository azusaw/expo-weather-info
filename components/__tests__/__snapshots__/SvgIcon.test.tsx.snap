// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SvgIcon it renders SvgIcon successfully 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight={100}
  bbWidth={100}
  fill="#888"
  focusable={false}
  height={100}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 100,
        "width": 100,
      },
    ]
  }
  vbHeight={570}
  vbWidth={570}
  width={100}
>
  <RNSVGGroup
    fill={
      {
        "payload": 4287137928,
        "type": 0,
      }
    }
    propList={
      [
        "fill",
      ]
    }
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      matrix={
        [
          1,
          0,
          0,
          1,
          93,
          29,
        ]
      }
    >
      <RNSVGPath
        d="M192 512C86 512 0 426 0 320 0 228.8 130.2 57.7 166.6 11.7c6-7.5 14.9-11.7 24.5-11.7h1.8c9.6 0 18.5 4.2 24.5 11.7C253.8 57.7 384 228.8 384 320c0 106-86 192-192 192M96 336c0-8.8-7.2-16-16-16s-16 7.2-16 16c0 61.9 50.1 112 112 112 8.8 0 16-7.2 16-16s-7.2-16-16-16c-44.2 0-80-35.8-80-80"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      />
    </RNSVGGroup>
  </RNSVGGroup>
</RNSVGSvgView>
`;
